Description: Change to work with Debian's maxima-sage packages
 Most changes are to find the differently named binary.
Author: Tobias Hansen <thansen@debian.org>

--- a/sage/src/sage/interfaces/maxima.py
+++ b/sage/src/sage/interfaces/maxima.py
@@ -546,7 +546,7 @@
         Expect.__init__(self,
                         name = 'maxima',
                         prompt = r'\(\%i[0-9]+\) ',
-                        command = 'maxima -p "{0}"'.format(STARTUP),
+                        command = 'maxima-sage -p "{0}"'.format(STARTUP),
                         env = {'TMPDIR': str(ECL_TMP)},
                         script_subdirectory = script_subdirectory,
                         restart_on_ctrlc = False,
@@ -786,7 +786,7 @@
         if not wait_for_prompt:
             return
         # line_echo sometimes has randomly inserted terminal echo in front #15811
-        assert line_echo.strip().endswith(line.strip()), 'mismatch:\n' + line_echo + line
+        #assert line_echo.strip().endswith(line.strip()), 'mismatch:\n' + line_echo + line
 
         self._expect_expr(self._display_prompt)
         out = self._before()  # input echo + output prompt + output
--- a/sage/src/sage/interfaces/maxima_abstract.py
+++ b/sage/src/sage/interfaces/maxima_abstract.py
@@ -168,7 +168,7 @@
             -- Function: gcd (<p_1>, <p_2>, <x_1>, ...)
             ...
         """
-        cmd = 'maxima --very-quiet --batch-string="%s(%s);" '%(command, s)
+        cmd = 'maxima-sage --very-quiet --batch-string="%s(%s);" '%(command, s)
         if sage.server.support.EMBEDDED_MODE:
             cmd += '< /dev/null'
 
@@ -2209,7 +2209,7 @@
         sage: maxima_version()  # random
         '5.41.0'
     """
-    with os.popen('maxima --version') as p:
+    with os.popen('maxima-sage --version') as p:
         return p.read().split()[-1]
 
 
@@ -2227,4 +2227,4 @@
     from sage.repl.rich_output.display_manager import get_display_manager
     if not get_display_manager().is_in_terminal():
         raise RuntimeError('Can use the console only in the terminal. Try %%maxima magics instead.')
-    os.system('maxima')
+    os.system('maxima-sage')
--- a/sage/src/sage/interfaces/expect.py
+++ b/sage/src/sage/interfaces/expect.py
@@ -610,7 +610,7 @@
 
             sage: a = maxima('y')
             sage: maxima.quit(verbose=True)
-            Exiting Maxima with PID ... running .../bin/maxima ...
+            Exiting Maxima with PID ... running .../bin/maxima-sage ...
             sage: a._check_valid()
             Traceback (most recent call last):
             ...
--- a/sage/src/bin/sage
+++ b/sage/src/bin/sage
@@ -578,7 +578,7 @@
 
 if [ "$1" = '-maxima' -o "$1" = '--maxima' ]; then
     shift
-    exec maxima "$@"
+    exec maxima-sage "$@"
 fi
 
 if [ "$1" = '-mwrank' -o "$1" = '--mwrank' ]; then
--- a/sage/src/doc/en/constructions/plotting.rst
+++ b/sage/src/doc/en/constructions/plotting.rst
@@ -214,7 +214,7 @@
 ::
 
     sage: maxima.eval('load("plotdf");')
-    '".../share/maxima/.../share/dynamics/plotdf.lisp"'
+    '".../share/maxima.../share/dynamics/plotdf.lisp"'
     sage: maxima.eval('plotdf(x+y,[trajectory_at,2,-0.1]); ')  # not tested
 
 This plots a direction field (the plotdf Maxima package was also
